README.md

```diff
@@ -2,30 +2,344 @@
 
 ## Learning Goals
 
-- Use terminal operations on Streams
+- Use methods that perform a terminal operation on a `Stream` object.
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+A stream pipeline consists of zero or more intermediate operations followed by a terminal operation.
+Each intermediate operation returns a new stream that is consumed by the next operation in the pipeline.
+A terminal operation ends a pipeline and may traverse a stream to produce a result or a side-effect. 
+A stream is consumed and can no longer be used after the terminal operation.
+A new stream must be created to traverse the same data source again.
 
-## Topic 1
+## count
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+`long	count()`
 
-## Topic 2
+The `count` method returns a `long` representing the number of elements in the stream.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+```java
+import java.util.List;
+import java.util.stream.Stream;
 
-## Conclusion
+public class Example {
+    public static void main(String[] args) {
+        List<Integer> numbers = List.of(-50, 20, 12, 4, -9);
+        long numCount = numbers.stream().count();
+        System.out.println(numCount); //5
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+        Stream<String> greetStream = Stream.of("hello", "hi there");
+        System.out.println(greetStream.count()); //2
+    }
+}
+```
+
+## max and min
+
+`Optional<T>	max(Comparator<? super T> comparator)`  
+`Optional<T>	min(Comparator<? super T> comparator)`
+
+The `max` method takes a `Comparator` as a parameter and returns a `Optional<T>` representing
+the maximum element of the stream according to the provided Comparator.
+
+- `T` is the type of the elements in the stream's data source.
+- The `Comparator` corresponds to the data source type `T`, i.e. `Integer::compare` for an `Integer` array or list.
+- The return type is `Optional` since the operation could be called on an empty collection, in which case there is no max value.
+
+In the example below,  the `max` function returns the 
+maximum value `55` wrapped in an `Optional` object
+when called on a non-empty list `ages`.
+The `max` function returns `Optional.empty` when
+called on the empty list `prices`.
+
+```java
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class Example {
+  public static void main(String[] args) {
+    List<Integer> ages = List.of(12, 55, 37, 9);
+    //List is not empty, a maximum value is returned as Optional[55]
```

Comment:
Just to add a little more readability, could we put a new line here in between `List<Integer> ages = ...` and the comment?

Reply:
ok, I added blank line.